crates/egui/src/context.rs

```diff
@@ -1907,60 +1910,95 @@ impl Context {
 impl Context {
     /// Associate some static bytes with a `uri`.
     ///
-    /// The same `uri` may be passed to [`Ui::image2`] later to load the bytes as an image.
-    pub fn include_static_bytes(&self, uri: &'static str, bytes: &'static [u8]) {
-        self.read(|ctx| ctx.loaders.include.insert_static(uri, bytes));
+    /// The same `uri` may be passed to [`Ui::image`] later to load the bytes as an image.
+    pub fn include_bytes(&self, uri: &'static str, bytes: impl Into<Bytes>) {
+        self.loaders().include.insert(uri, bytes.into());
     }
 
-    /// Associate some bytes with a `uri`.
-    ///
-    /// The same `uri` may be passed to [`Ui::image2`] later to load the bytes as an image.
-    pub fn include_bytes(&self, uri: &'static str, bytes: impl Into<Arc<[u8]>>) {
-        self.read(|ctx| ctx.loaders.include.insert_shared(uri, bytes));
+    /// Returns `true` if the chain of bytes, image, or texture loaders
+    /// contains a loader with the given `id`.
+    pub fn is_loader_installed(&self, id: &str) -> bool {
+        let loaders = self.loaders();
+        if loaders.bytes.lock().iter().any(|loader| loader.id() == id) {
+            return true;
+        }
+        if loaders.image.lock().iter().any(|loader| loader.id() == id) {
+            return true;
+        }
+        if loaders
+            .texture
+            .lock()
+            .iter()
+            .any(|loader| loader.id() == id)
+        {
+            return true;
+        }
+        false
```

Comment:
I wanted to keep the individual loader `Vec`s locked for as short of a time as possible. I refactored it a bit so that the intermediate `.lock().iter().any(...)` are stored in variables, which drops the lock early, and makes this a bit cleaner.